tensorboard/webapp/feature_flag/views/feature_flag_page_container_test.ts

```diff
@@ -0,0 +1,130 @@
+/* Copyright 2022 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+import {CommonModule} from '@angular/common';
+import {TestBed} from '@angular/core/testing';
+import {Store} from '@ngrx/store';
+import {MockStore, provideMockStore} from '@ngrx/store/testing';
+import {State} from '../../app_state';
+import {
+  allFeatureFlagOverridesReset,
+  featureFlagOverrideChanged,
+  featureFlagOverridesReset,
+} from '../actions/feature_flag_actions';
+import {
+  getDefaultFeatureFlags,
+  getOverriddenFeatureFlags,
+} from '../store/feature_flag_selectors';
+import {FeatureFlags} from '../types';
+import {FeatureFlagPageModule} from './feature_flag_module';
+import {
+  FeatureFlagPageContainer,
+  TEST_ONLY,
+} from './feature_flag_page_container';
+import {FeatureFlagOverrideStatus} from './types';
+
+describe('feature_flag_page_container', () => {
+  let store: MockStore<State>;
+  let dispatchSpy: jasmine.Spy;
+
+  beforeEach(async () => {
+    await TestBed.configureTestingModule({
+      declarations: [FeatureFlagPageContainer],
+      imports: [CommonModule, FeatureFlagPageModule],
+      providers: [provideMockStore(), FeatureFlagPageContainer],
+    }).compileComponents();
+    store = TestBed.inject<Store<State>>(Store) as MockStore<State>;
+    dispatchSpy = spyOn(store, 'dispatch');
+  });
+
+  function getComponent() {
+    return TestBed.createComponent(FeatureFlagPageContainer).componentInstance;
+  }
+
+  describe('onFlagChanged', () => {
+    it('creates override status is set to enabled or disabled not set', () => {
+      store.overrideSelector(getDefaultFeatureFlags, {} as FeatureFlags);
+      store.overrideSelector(getOverriddenFeatureFlags, {});
+      const component = getComponent();
+      component.onFlagChanged({
```

Comment:
Again in the context of a single test file for each container/component combination, we generally prefer to test dispatching actions starting at the UI. That is: setup your ngrx selectors as appropriate, render the container/component/UI, interact with an element, verify the action is fired.

Reply:
I believe I have done this correctly.

Reply:
I've merged the files and refactored the tests to move away from `componentInstance` except in the instance of `formatFlagValue` tests which I believe are significantly simpler as is.